packages/amplify-graphql-function-transformer/src/__tests__/amplify-graphql-function-transformer.test.ts

```diff
@@ -369,3 +371,318 @@ test('includes auth info in stash', () => {
   });
   expect(out.stacks.FunctionDirectiveStack.Resources!.QuerymyFunctionResolver.Properties.RequestMappingTemplate).toMatchSnapshot();
 });
+
+describe('Event invocation type', () => {
+  test('Defined on Query succeeds', () => {
+    const schema = `
+      type Query {
+        asyncStuff(msg: String): EventInvocationResponse @function(name: "asyncstuff-\${env}", invocationType: Event)
+      }
+
+      type EventInvocationResponse {
+        success: Boolean!
+      }
+    `;
+
+    const out = testTransform({ schema, transformers: [new FunctionTransformer()] });
+    expect(out).toBeDefined();
+    parse(out.schema);
+    expect(out.stacks).toBeDefined();
+
+    const stack = out.stacks.FunctionDirectiveStack;
+    expect(stack).toBeDefined();
+
+    const resolvers = out.resolvers;
+    expect(resolvers).toBeDefined();
+    expect(resolvers).toMatchSnapshot();
+  });
+
+  test('Defined on Mutation succeeds', () => {
+    const schema = `
+    type Mutation {
+      asyncStuff(msg: String): EventInvocationResponse @function(name: "asyncstuff-\${env}", invocationType: Event)
+    }
+
+    type EventInvocationResponse {
+      success: Boolean!
+    }
+  `;
+
+    const out = testTransform({ schema, transformers: [new FunctionTransformer()] });
+    expect(out).toBeDefined();
+
+    parse(out.schema);
+    expect(out.schema).toBeDefined();
+    expect(out.schema).toMatchSnapshot();
+
+    expect(out.stacks).toBeDefined();
+    const stack = out.stacks.FunctionDirectiveStack;
+    expect(stack).toBeDefined();
+
+    const resolvers = out.resolvers;
+    expect(resolvers).toBeDefined();
+    expect(resolvers).toMatchSnapshot();
+  });
+
+  test('event invocation fails on non query or mutation parent type', () => {
+    const schema = `
+    type Foo @model {
+      asyncStuff(msg: String): EventInvocationResponse @function(name: "asyncstuff-\${env}", invocationType: Event)
+    }
+
+    type EventInvocationResponse {
+      success: Boolean!
+    }
+    `;
+
+    expect(() => testTransform({ schema, transformers: [new ModelTransformer(), new FunctionTransformer()] })).toThrowError(
+      "@function definition with 'invocationType: Event' must be defined on Query or Mutation field.",
+    );
+  });
+
+  describe('Chained directives', () => {
+    test('chained @function -- RequestResponse > Event succeeds with EventInvocationResponse', () => {
+      const schema = `
+        type Mutation {
+          asyncStuff(msg: String): EventInvocationResponse
+          @function(name: "asyncstuff2-\${env}")
+          @function(name: "asyncstuff-\${env}", invocationType: Event)
+        }
+        type EventInvocationResponse {
+          success: Boolean!
+        }
+      `;
+      const out = testTransform({ schema, transformers: [new FunctionTransformer()] });
+      expect(out).toBeDefined();
```

Comment:
Here and other success tests: Do we need to assert the execution order & return value of the resolver? Or like, a snapshot? :)

Reply:
Great point. Addressed in https://github.com/aws-amplify/amplify-category-api/pull/2868